torch_frame/data/tensor_frame.py

```diff
@@ -207,13 +236,37 @@ def __copy__(self) -> 'TensorFrame':
     # Device Transfer #########################################################
 
     def to(self, *args, **kwargs):
-        return self._apply(lambda x: x.to(*args, **kwargs))
+        def fn(x):
+            if isinstance(x, dict):
+                for key in x:
+                    x[key] = x[key].to(*args, **kwargs)
+            else:
+                x = x.to(*args, **kwargs)
+            return x
+
+        return self._apply(fn)
 
     def cpu(self, *args, **kwargs):
-        return self._apply(lambda x: x.cpu(*args, **kwargs))
+        def fn(x):
+            if isinstance(x, dict):
+                for key in x:
+                    x[key] = x[key].cpu(*args, **kwargs)
+            else:
+                x = x.cpu(*args, **kwargs)
+            return x
+
+        return self._apply(fn)
 
     def cuda(self, *args, **kwargs):
-        return self._apply(lambda x: x.cuda(*args, **kwargs))
+        def fn(x):
+            if isinstance(x, dict):
+                for key in x:
+                    x[key] = x[key].cuda(*args, **kwargs)
+            else:
+                x = x.cuda(*args, **kwargs)
+            return x
+
+        return self._apply(fn)
```

Comment:
The logic seems to be repetitive. Can you just call `to('cuda')` here? Same with `to('cpu')`